kodilla-good-patterns/src/main/java/com/kodilla/good/patterns/challenges/service/OrderService.java

```diff
@@ -0,0 +1,34 @@
+package com.kodilla.good.patterns.challenges.service;
+
+import com.kodilla.good.patterns.challenges.entity.Order;
+import com.kodilla.good.patterns.challenges.entity.OrderPosition;
+import com.kodilla.good.patterns.challenges.infrastructure.ComponentRegistry;
+import com.kodilla.good.patterns.challenges.model.OrderRequest;
+import com.kodilla.good.patterns.challenges.repository.OrderRepository;
+
+import java.util.Optional;
+
+public class OrderService {
+    private final OrderPositionService orderPositionService = ComponentRegistry.getOrderPositionService();
+    private final OrderRepository orderRepository = ComponentRegistry.getOrderRepository();
+
+    public Optional<Order> order(OrderRequest orderRequest) {
+        Order order = new Order(orderRequest.getOrdersPositions(), orderRequest.getAddress());
+        if (validateOrder(order)) {
+            String id = orderRepository.save(order).trim();
+            if (!id.equals("")) {
+                return Optional.of(order);
+            }
```

Comment:
z ta mechanika jest cos nie tak:
- dlaczego konieczne jest `trim`?
- klaczego id moze byc empty stringiem?

Reply:
Nie było to niezbędne